src/OpenLoco/src/Paint/Paint.cpp

```diff
@@ -189,6 +192,466 @@ namespace OpenLoco::Paint
+        enumValue(World::Track::TrackId::rightCurveSmall),
+        enumValue(World::Track::TrackId::leftCurveSmall),
+        enumValue(World::Track::TrackId::rightCurve),
+        enumValue(World::Track::TrackId::leftCurve),
+        enumValue(World::Track::TrackId::rightCurveLarge),
+        enumValue(World::Track::TrackId::leftCurveLarge),
+        enumValue(World::Track::TrackId::diagonalRightCurveLarge),
+        enumValue(World::Track::TrackId::diagonalLeftCurveLarge),
+        enumValue(World::Track::TrackId::sBendLeft),
+        enumValue(World::Track::TrackId::sBendRight),
+        enumValue(World::Track::TrackId::straightSlopeUp),
+        enumValue(World::Track::TrackId::straightSlopeDown),
+        enumValue(World::Track::TrackId::straightSteepSlopeUp),
+        enumValue(World::Track::TrackId::straightSteepSlopeDown),
+        enumValue(World::Track::TrackId::rightCurveSmallSlopeUp),
+        enumValue(World::Track::TrackId::rightCurveSmallSlopeDown),
+        enumValue(World::Track::TrackId::leftCurveSmallSlopeUp),
+        enumValue(World::Track::TrackId::leftCurveSmallSlopeDown),
+        enumValue(World::Track::TrackId::rightCurveSmallSteepSlopeUp),
+        enumValue(World::Track::TrackId::rightCurveSmallSteepSlopeDown),
+        enumValue(World::Track::TrackId::leftCurveSmallSteepSlopeUp),
+        enumValue(World::Track::TrackId::leftCurveSmallSteepSlopeDown),
+    };
+
+    void PaintSession::printTP()
+    {
+
+        std::vector<std::string> tppNames;
+        for (auto trackId : kTrackOrder)
+        {
+            std::vector<std::string> ppNames;
+            auto& track = tp.tracks[trackId];
+            for (auto index = 0U; index < track.size(); ++index)
+            {
+                auto& ti = track[index];
+                if (std::holds_alternative<TestPaint::Track3>(ti))
+                {
+                    auto& t3Ct = std::get<TestPaint::Track3>(ti);
+                    ppNames.push_back(printTrack3Alt(t3Ct, trackId, index));
+                }
+                else if (std::holds_alternative<TestPaint::Track1>(ti))
+                {
+                    auto& t3Ct = std::get<TestPaint::Track1>(ti);
+                    ppNames.push_back(printTrack1Alt(t3Ct, trackId, index));
+                }
+            }
+            if (!track.empty())
+            {
+                std::string tppName = fmt::format("k{}TPP", _trackIdNames[trackId]);
+                tppNames.push_back(tppName);
+                std::cout << fmt::format("constexpr std::array<TrackPaintPiece, {}> {} = {{\n", track.size(), tppName);
+                for (auto& ppName : ppNames)
+                {
+                    std::cout << fmt::format("    {},\n", ppName);
+                }
+                std::cout << "};\n\n";
+            }
+        }
+        std::cout << fmt::format("constexpr std::array<std::span<const TrackPaintPiece>, {}> kTrackPaintParts = {{\n", tppNames.size());
+        for (auto& tppName : tppNames)
```

Comment:
This will be out of order need to remember to change it back from kTrackOrder to trackid order